bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -257,6 +258,7 @@ public void dispose() {
     private void createChannelsGeneral() {
         final String channelGroup = "general";
         if (batteryConfigured) {
+            createChannel(CHANNEL_BATTERY_CAPACITY, channelGroup, CHANNEL_TYPE_UID_BATTERY_CAPACITY, "Number:Energy");
```

Comment:
Just a comment for future consideration: Instead of dynamically creating channels, they could be statically declared in XML and then removed when not needed. See for example #13275. The reason for doing so would be simplification.